pkg/pillar/cmd/zfsmanager/handlesnapshotmgr.go

```diff
@@ -0,0 +1,312 @@
+package zfsmanager
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	libzfs "github.com/bicomsystems/go-libzfs"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	"github.com/lf-edge/eve/pkg/pillar/zfs"
+)
+
+func snapshotStatus(zvolName string, snapshotConfig types.ZfsSnapshotConfig) (types.ZfsSnapshotStatus, error) {
+	volumeConfig := types.ZfsSnapshotStatus{}
+	var dataset libzfs.Dataset
+	var err error
+	if !zfs.DoesSnapshotExist(zvolName, snapshotConfig.UUID) {
+		// Сreate a new snapshot status. This is the cmd to take the snapshot
+		// Before this operation, you should execute the necessary code or
+		// make sure that all processes and conditions for creating a
+		// snapshot in EVE are created
```

Comment:
Seems we should also check for deleted state in previous status we have, and if we have deleted, we should not create snapshot again.